### PR TITLE
nixos/gitlab: Fix databaseCreateLocally evaluation and operation

### PR DESCRIPTION
**Pull request by talyz**
###### Motivation for this change
Currently, as long as the `services.gitlab.databaseCreateLocally` option is set to `true`, the default, we create a local database, even if the `services.gitlab.databaseHost` option is set, indicating that we're not going to connect to the local database. We also assert that `services.gitlab.user` and `services.gitlab.databaseUsername` are equal, which in actuality only matters when `services.gitlab.databaseHost` is set to its default value `""`. This issue was raised in #68841.

Even worse, setting `services.gitlab.databaseCreateLocally` to `false` results in an evaluation failure unless `services.postgresql.enable` is set to `true` either manually or by another module.

This fixes both of these issues and needs to be ported to 19.09.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @aanderse @flokli @globin @mawis

### CODE REVIEW

File: nixos/modules/services/misc/gitlab.nix [hunk 1]
```diff
@@ -7,6 +7,11 @@ let
 
   ruby = cfg.packages.gitlab.ruby;
 
+  postgresqlPackage = if config.services.postgresql.enable == true then
```
Comment (globin) [inline nixos/modules/services/misc/gitlab.nix]:
```suggestion
  postgresqlPackage = if config.services.postgresql.enable then
```

**Review by globin — APPROVED** [review]
Fine except for style nitpick

File: nixos/modules/services/misc/gitlab.nix (hunk 1 above)
Comment (talyz) [reply nixos/modules/services/misc/gitlab.nix]: Ah, right, of course..

**Comment by petabyteboy** [comment]
Can this be merged? @globin

**Pull request merged**